src/components/Docs/components/ConnectedChainsList.tsx

```diff
@@ -0,0 +1,188 @@
+import { useEffect, useMemo, useState } from "react";
+
+import { LoadingTable, NetworkTypeTabs, networkTypeTabs, rpcByNetworkType } from "~/components/shared";
+
+type Chain = {
+  chain_id: string;
+  chain_name: string;
+  network: string;
+  network_type: string;
+  vm: string;
+  consensus: string;
+  is_external: boolean;
+  cctx_gateway: string;
+  name: string;
+};
+
+type ForeignCoin = {
+  zrc20_contract_address: string;
+  asset: string;
+  foreign_chain_id: string;
+  decimals: number;
+  name: string;
+  symbol: string;
+  coin_type: string;
+};
+
+type CoinsData = {
+  foreignCoins: ForeignCoin[];
+};
+
+type ChainsData = {
+  chains: Chain[];
+};
+
+const CHAINS = "/zeta-chain/observer/supportedChains";
+const COINS = "/zeta-chain/fungible/foreign_coins";
+
+const formatString = (str: string) => {
+  return str
+    .split("_")
+    .map((word: string) =>
+      word.length <= 3 ? word.toUpperCase() : word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()
+    )
+    .join(" ");
+};
+
+export const ConnectedChainsList = () => {
+  const [mainnetChains, setMainnetChains] = useState<Chain[]>([]);
+  const [testnetChains, setTestnetChains] = useState<Chain[]>([]);
+  const [tokens, setTokens] = useState<ForeignCoin[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [activeTab, setActiveTab] = useState(networkTypeTabs[0]);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    const fetchData = async () => {
+      try {
+        const CHAINS_URL = `${rpcByNetworkType[activeTab.networkType]}${CHAINS}`;
+        const COINS_URL = `${rpcByNetworkType[activeTab.networkType]}${COINS}`;
+
+        const [responseChains, responseTokens] = await Promise.all([fetch(CHAINS_URL), fetch(COINS_URL)]);
+
+        const chainsData: ChainsData = await responseChains.json();
+        const tokensData: CoinsData = await responseTokens.json();
```

Comment:
Nit small improvement

```suggestion
        const [chainsData, tokensData] = await Promise.all<[ChainsData, CoinsData]>([
          fetch(CHAINS_URL).then((res) => res.json()),
          fetch(COINS_URL).then((res) => res.json()),
        ]);
```

Reply:
1758e53

Reply:
1758e53